src/notification/sms.ts

```diff
@@ -19,41 +23,57 @@ const transporter = nodemailer.createTransport({
 });
 
 export function sendSMS(link: Link, store: Store) {
-	if (phone.number) {
+	if (phone.number.length > 0 && phone.carrier.length > 0) {
 		logger.debug('↗ sending sms');
-		const carrier = phone.carrier;
-
-		if (carrier && phone.availableCarriers.has(carrier)) {
-			const mailOptions: Mail.Options = {
-				attachments: link.screenshot ? [
-					{
-						filename: link.screenshot,
-						path: `./${link.screenshot}`
-					}
-				] : undefined,
-				from: email.username,
-				subject: Print.inStock(link, store, false, true),
-				text: link.cartUrl ? link.cartUrl : link.url,
-				to: generateAddress()
-			};
-
-			transporter.sendMail(mailOptions, error => {
-				if (error) {
-					logger.error('✖ couldn\'t send sms', error);
-				} else {
-					logger.info('✔ sms sent');
+		const carrier = phone.carrier[0].trim();
+
+		const mailOptions: Mail.Options = {
+			attachments: link.screenshot ? [
+				{
+					filename: link.screenshot,
+					path: `./${link.screenshot}`
+				}
+			] : undefined,
+			from: email.username,
+			subject: Print.inStock(link, store, false, true),
+			text: link.cartUrl ? link.cartUrl : link.url
+		};
+
+		if (phone.carrier.length === 1) {
+			if (carrier && phone.availableCarriers.has(carrier)) {
+				for (const phoneNumber of phone.number) {
+					mailOptions.to = generateAddress(phone.carrier[0], phoneNumber);
+					sendMail(mailOptions);
 				}
-			});
+			}
+		} else {
+			for (let i = 0; i < phone.number.length; i++) {
+				if (phone.carrier[i] && phone.availableCarriers.has(phone.carrier[i].trim())) {
+					mailOptions.to = generateAddress(phone.carrier[i], phone.number[i]);
+					sendMail(mailOptions);
+				}
+			}
```

Comment:
Let's keep the original logic and just iterate through `carrier` or `number` and use that index for one or the other since we already do a check for equality in size.